waku/peer.go

```diff
@@ -94,17 +100,21 @@ func (p *Peer) handshake() error {
 	// Send the handshake status message asynchronously
 	errc := make(chan error, 1)
 	isLightNode := p.host.LightClientMode()
-	isRestrictedLightNodeConnection := p.host.LightClientModeConnectionRestricted()
+	confirmationsEnabled := p.host.ConfirmationsEnabled()
+	rateLimits := p.host.RateLimits()
 	go func() {
 		opts := statusOptions{
 			BloomFilter:          p.host.BloomFilter(),
-			LightNodeEnabled:     isLightNode,
-			ConfirmationsEnabled: p.host.ConfirmationsEnabled(),
-			RateLimits:           p.host.RateLimits(),
-			TopicInterest:        nil,
+			TopicInterest:        p.host.TopicInterest(),
+			RateLimits:           &rateLimits,
+			LightNodeEnabled:     &isLightNode,
+			ConfirmationsEnabled: &confirmationsEnabled,
 		}
-		opts.SetPoWRequirementFromF(p.host.MinPow())
-		errc <- p2p.SendItems(p.ws, statusCode, ProtocolVersion, opts)
+		pow := p.host.MinPow()
+		if pow != 0 {
+			opts.SetPoWRequirementFromF(pow)
```

Comment:
currently that's not an issue, as `topicInterest`and bloom are never used in conjunction (setting one sets the other to `nil`), but we can be explicit about it

Reply:
Right but when I look at this I would wonder: "I use bloom mode but in the code I still propagate `topic-interest`. Why is it so? Shouldn't I only send my bloom filter" :) 

Also, when receiving options we do `peerOptions.TopicInterest != nil`. But if we always send `topic-interest`, after RLP decoding it won't be `nil`, it will be an empty slice so we never reach `else if peerOptions.BloomFilter != nil `. Did I miss anything?